script.js

```diff
@@ -1,12 +1,574 @@
-//You can edit ALL of the code here
-function setup() {
-  const allEpisodes = getAllEpisodes();
-  makePageForEpisodes(allEpisodes);
+//Setting global variables
+
+let allShows;
+let allEpisodes;
+let isShowsListing = true;
+let episodeSelectContainer;
+let searchContainer;
+let showSearchInput;
+let showSelect;
+let searchInput;
+let episodeSelect;
+let showSearchContainer;
+
+
+
+async function setup() {
+  allShows = await getAllShows();
+  const selectedShowId = allShows[0].id; // Selecting the first show by default
+  allEpisodes = await getEpisodes(selectedShowId);
+  createHeader();
+  createShowSearch();
+  showSelectField(allShows);
+  showListing();
+  createFooter();
+}
+
+// Fetching episodes for shows
+function getEpisodes(showId) {
+  return fetch(`https://api.tvmaze.com/shows/${showId}/episodes`)
+    .then(response => {
+      if (!response.ok) {
+        throw new Error('Error fetching episodes');
+      }
+      return response.json();
+    })
+    .catch(error => {
+      console.error('Error fetching episodes:', error);
+    });
+}
+
+
+function backToShows() {
+let rootElem = document.getElementById("root");
+
+let showsButton = document.createElement("button");
+showsButton.classList.add("backToShows");
+
+showsButton.textContent = "Back to Shows";
+
+showsButton.addEventListener("click", function () {
+  showListing(); 
+});
+rootElem.appendChild(showsButton);
+
+}
+
+function searchShows(searchValue, showSelect) {
```

Comment:
You have defined the parameters searchValue and showSelect. When you call the function on line 377 you do not pass in any arguments. Therefore, I think these are not needed?